src/main/java/duke/command/DukeCommand.java

```diff
@@ -0,0 +1,115 @@
+package duke.command;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.exceptions.DukeException;
+import duke.task.ToDo;
+
+import java.io.FileWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class DukeCommand extends Duke {
+
+    protected static int taskCount = 0;
+    protected static String[] description;
+    protected static String[] separate;
+    public static boolean isPlural = false;
+
+    public static void readFile(String filePath, ArrayList<Task> tasks) throws FileNotFoundException {
+        ArrayList<String> fileContents = new ArrayList<>();
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            fileContents.add(s.nextLine());
+        }
+        for (int i = 0; i < fileContents.size(); i++) {
+            if (fileContents.get(i).startsWith("T")) {
+                description = fileContents.get(i).split("]");
+                tasks.add(new ToDo(description[1].trim()));
+                if (fileContents.get(i).contains("[X]")) {
+                    tasks.get(i).markAsDone();
+                }
+            } else if (fileContents.get(i).startsWith("D")) {
+                int by = fileContents.get(i).indexOf("by:");
+                separate = fileContents.get(i).split("by:");
+                description = separate[0].trim().split("]");
+                String dueDate = separate[1].trim();
+                tasks.add(new Deadline(description[1].trim(), dueDate));
+                if (fileContents.get(i).contains("[X]")) {
+                    tasks.get(i).markAsDone();
+                }
+            } else if (fileContents.get(i).startsWith("E")) {
+                int at = fileContents.get(i).indexOf("at:");
+                separate = fileContents.get(i).split("at:");
+                description = separate[0].trim().split("]");
+                String eventTiming = separate[1].trim();
+                tasks.add(new Event(description[1].trim(), eventTiming));
+                if (fileContents.get(i).contains("[X]")) {
+                    tasks.get(i).markAsDone();
+                }
+            }
```

Comment:
You can refactor this code to a method with parameters. That way you improve the SLAP in this method.